dozer-admin/src/services/application_service.rs

```diff
@@ -300,4 +321,120 @@ impl AppService {
             },
         )
     }
+
+    pub fn list_files(&self) -> Result<ListFilesResponse, ErrorResponse> {
+        let mut files = vec![];
+        let files_glob = glob("*.yaml").map_err(|e| ErrorResponse {
+            message: e.to_string(),
+        })?;
+
+        for entry in files_glob {
+            match entry {
+                Ok(path) => {
+                    files.push(File {
+                        name: format!("{:?}", path.display()),
+                        content: fs::read_to_string(path).map_err(|e| ErrorResponse {
+                            message: e.to_string(),
+                        })?,
+                    });
+                }
+                Err(e) => {
+                    return Err(ErrorResponse {
+                        message: e.to_string(),
+                    })
+                }
+            }
+        }
+
+        Ok(ListFilesResponse { files })
+    }
+
+    pub fn save_files(
+        &self,
+        request: SaveFilesRequest,
+    ) -> Result<SaveFilesResponse, ErrorResponse> {
+        for file in request.files {
+            let mut fs_file = fs::File::create(file.name).unwrap();
+            fs_file.write_all(file.content.as_bytes()).unwrap();
+        }
+
+        Ok(SaveFilesResponse {})
+    }
+
+    pub async fn read_logs(tx: tokio::sync::mpsc::Sender<Result<LogMessage, Status>>) {
+        let mut interval = interval(Duration::from_millis(1000));
+        let mut position = 0;
+
+        loop {
+            let file_result = fs::File::open("./log/dozer.log");
+            if let Ok(mut file) = file_result {
+                let seek_result = file.seek(SeekFrom::Start(position));
+
+                match seek_result {
+                    Ok(_) => {
+                        if let Ok(metadata) = file.metadata() {
+                            position = metadata.len();
+                            let reader = BufReader::new(file);
+                            for message in reader.lines().flatten() {
+                                let log_message = LogMessage { message };
+
+                                if (tx.send(Ok(log_message)).await).is_err() {
+                                    // receiver dropped
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                    Err(e) => {
+                        warn!("Seek error: {:?}", e);
+                    }
+                }
+            }
+
+            interval.tick().await;
+        }
+    }
+
+    pub async fn stream_status_update(
+        request: StatusUpdateRequest,
+        tx: tokio::sync::mpsc::Sender<Result<StatusUpdate, Status>>,
+    ) {
+        let mut retries_left = 100;
```

Comment:
Is 100s timeout too long? And should we send a status update if can't connect to dozer?

Reply:
I haven't added anything related to connection for now, because I don't know how we would show it in admin UI. Once, we have more details, I can make these changes.